Hyper/MarchingCubes/ChunkWorker.cs

```diff
@@ -0,0 +1,114 @@
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Reflection.Metadata;
+using System.Threading;
+using Hyper.Collisions;
+using Hyper.MarchingCubes;
+using Hyper.Meshes;
+using OpenTK.Mathematics;
+using Hyper.Collisions.Bepu;
+
+namespace Hyper
+{
+    internal class ChunkWorker
+    {
+        private ConcurrentDictionary<Vector3i, Guid> _neededChunkPositions;
+        private ConcurrentDictionary<Vector3i, Guid> _existingChunkPositions;
```

Comment:
I don't get why we use GUID when Vector3i are unique and we could just use them

Reply:
I can easily change this to concurrent set. i was thinking about a situation where we maybe had a conflict between freshly generated chunk and a chunk stored in a database

Reply:
![image](https://github.com/Non-Euclidean-World/Hyper/assets/45049508/f9877743-4292-42a2-88d6-f30bc9d0b130)
The only problem is that there is no concurrent set to my knowledge.
https://github.com/dotnet/runtime/issues/16443
I only took a glance.

Reply:
If we don't use values at all, then could we just have `ConcurrentDictionary<Vector3i, int>` and aways add `(vec, 0)` for readability?

Reply:
Just as a side note, I looked up if there's some sort of "throw away" type in C# that could be used here., like in C++ there's `nullptr_t`.  And it turns out that in C# the `null` literal has its own type! The only thing is "There is no way to explicitly write the null type" (https://stackoverflow.com/a/8204622/15767108)